mlflow/data/numpy_dataset.py

```diff
@@ -25,16 +25,17 @@ class NumpyDataset(Dataset, PyFuncConvertibleDatasetMixin):
 
     def __init__(
         self,
-        features: Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]],
```

Comment:
MLflow schemas don't support list of ndarray. Users can work around this by naming their tensors (i.e. using dicts)